src/EnergyPlus/api/runtime.cc

```diff
@@ -83,147 +90,180 @@ void issueText(const char * message) {
     EnergyPlus::ShowContinueError(message);
 }
 
-void registerProgressCallback(void (*f)(int const)) {
+void registerProgressCallback(EnergyPlusState EP_UNUSED(state), void (*f)(int const)) {
+    // auto *this_state = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
     EnergyPlus::DataGlobals::progressCallback = f;
 }
 
-void registerStdOutCallback(void (*f)(const char * message)) {
+void registerStdOutCallback(EnergyPlusState EP_UNUSED(state), void (*f)(const char * message)) {
     EnergyPlus::DataGlobals::messageCallback = f;
 }
 
-void callbackBeginNewEnvironment(std::function<void ()> f) {
-    EnergyPlus::PluginManagement::registerNewCallback(EnergyPlus::DataGlobals::emsCallFromBeginNewEvironment, f);
+void callbackBeginNewEnvironment(EnergyPlusState state, std::function<void (EnergyPlusState)> const &f) {
```

Comment:
As mentioned previously, state is now passed into the API when a client is registering a callback, and the callback function itself should now accept a void* argument (the state).